### PR TITLE
ENH: Added GARCH class and changed example

### PR DESCRIPTION
**Pull request by springcoil**
Enhancement added GARCH class and tested with example

### CODE REVIEW

File: pymc3/examples/garch_example.py [hunk 1]
```diff
@@ -42,14 +49,21 @@
 
     theta = T.sqrt(alpha0 + alpha1 * T.pow(r - mu, 2) + beta1 * T.pow(sigma1, 2))
```
Comment (twiecki) [inline pymc3/examples/garch_example.py:50]: Is theta still required?

**Comment by olafSmits** [comment]
This does not produce the GARCH model I'm afraid. You cannot vectorize the construction of the volatility process `sigma` -- it has to be defined in terms of a loop.

**Comment by springcoil** [comment]
Hmm - back to the drawing board
Any suggestions @olafSmits of how to use scan here.

File: pymc3/examples/garch_example.py (hunk 1 above)
Comment (springcoil) [inline pymc3/examples/garch_example.py:50]: Not sure, it might not be.

**Pull request closed without merging**